internal/services/containers/kubernetes_cluster_resource_test.go

```diff
@@ -923,3 +952,49 @@ resource "azurerm_kubernetes_cluster" "test" {
 }
   `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, controlPlaneVersion)
 }
+
+func (KubernetesClusterResource) upgradeOverrideSetting(data acceptance.TestData, isUpgradeOverrideSettingEnabled bool) string {
+
+	upgradeOverrideSetting := ""
+	if isUpgradeOverrideSettingEnabled {
+		upgradeOverrideSetting = `
+  upgrade_override_setting {
+	effective_until = "2024-01-01T00:00:00Z"
```

Comment:
Does it make sense to be testing a time in the past here? It's unlikely this will be representative of user input? Could we generate a value for this based on `time.Now()` plus some realistic duration?